### DISCUSSION TITLE
Wonder on Authentication in Adonis JS

### DISCUSSION THREAD

**Discussion by tuandungbrse** [adonisjs/core#3360] [Help] [answered]
Hi AdonisJS Team, thank you for the amazing framework!

I'm just getting started with Adonis JS and about to implement Authentication by API Tokens.

I'm still confused why the Opaque Token that is saved in the database is better than JWT. I mean, they both can be stolen, right?

**Comment by thetutlage** [comment, marked as answer]
Shared it many times earlier. https://github.com/adonisjs/core/discussions/2166#discussioncomment-307787, https://github.com/adonisjs/core/discussions/1365#discussioncomment-44972
